Source/WebKit/NetworkProcess/webrtc/NetworkRTCTCPSocketCocoa.mm

```diff
@@ -117,20 +118,21 @@ static inline void processIncomingData(RetainPtr<nw_connection_t>&& nwConnection
     m_nwConnection = createNWConnection(rtcProvider, hostName.c_str(), String::number(remoteAddress.port()).utf8().data(), isTLS, attributedBundleIdentifier, isFirstParty, isRelayDisabled, domain);
 
     nw_connection_set_queue(m_nwConnection.get(), tcpSocketQueue());
-    nw_connection_set_state_changed_handler(m_nwConnection.get(), makeBlockPtr([weakThis = WeakPtr { *this }, identifier = m_identifier, rtcProvider = Ref { rtcProvider }, connection = m_connection.copyRef()](nw_connection_state_t state, _Nullable nw_error_t error) {
+    nw_connection_set_state_changed_handler(m_nwConnection.get(), makeBlockPtr([weakNWConnection = WeakObjCPtr { m_nwConnection.get() }, identifier = m_identifier, rtcProvider = Ref { rtcProvider }, connection = m_connection.copyRef()](nw_connection_state_t state, _Nullable nw_error_t error) {
         switch (state) {
         case nw_connection_state_invalid:
         case nw_connection_state_waiting:
         case nw_connection_state_preparing:
             return;
         case nw_connection_state_ready:
-            rtcProvider->callOnRTCNetworkThread([weakThis, connection, identifier] {
-                if (!weakThis)
+            rtcProvider->callOnRTCNetworkThread([weakNWConnection, connection, identifier] {
+                RetainPtr nwConnection = weakNWConnection.get();
+                if (!nwConnection)
                     return;
-                auto path = adoptNS(nw_connection_copy_current_path(weakThis->m_nwConnection.get()));
-                auto interface = adoptNS(nw_path_copy_interface(path.get()));
-                if (auto* name = nw_interface_get_name(interface.get()))
-                    connection->send(Messages::LibWebRTCNetwork::SignalUsedInterface(identifier, String::fromUTF8(name)), 0);
+                RetainPtr path = adoptNS(nw_connection_copy_current_path(nwConnection.get()));
+                RetainPtr interface = adoptNS(nw_path_copy_interface(path.get()));
+                if (auto name = String::fromUTF8(nw_interface_get_name(interface.get())); !name.isNull())
```

Comment:
Nice improvement here with slightly better handling of the case where the UTF-8 conversion might fail.